xbmc/Application.cpp

```diff
@@ -4865,6 +4865,13 @@ bool CApplication::OnMessage(CGUIMessage& message)
       if (url.IsProtocol("plugin"))
         XFILE::CPluginDirectory::GetPluginResult(url.Get(), file);
 
+      // Don't queue if next media type is different from current one
+      if ((!file.IsVideo() && m_pPlayer->IsPlayingVideo()) ||
```

Comment:
Ok, the idea is sane but I am wondering if we need to change implementation slightly. The OnSomething methods are supposed to be called on an event. In this case it looks like that playback has not ended at this point, right? Any reason why you don't stop it explicitly?

(btw: our coding style defines curly brackets on next line.)